test/e2e/reconciler/partition/partitionset_test.go

```diff
@@ -0,0 +1,332 @@
+/*
+Copyright 2023 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package partition
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/davecgh/go-spew/spew"
+	"github.com/stretchr/testify/require"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/validation"
+	"k8s.io/apimachinery/pkg/util/wait"
+
+	"github.com/kcp-dev/kcp/pkg/apis/core"
+	corev1alpha1 "github.com/kcp-dev/kcp/pkg/apis/core/v1alpha1"
+	"github.com/kcp-dev/kcp/pkg/apis/third_party/conditions/util/conditions"
+	topologyv1alpha1 "github.com/kcp-dev/kcp/pkg/apis/topology/v1alpha1"
+	kcpclientset "github.com/kcp-dev/kcp/pkg/client/clientset/versioned/cluster"
+	"github.com/kcp-dev/kcp/test/e2e/framework"
+)
+
+func TestPartitionSet(t *testing.T) {
+	t.Parallel()
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	server := framework.PrivateKcpServer(t)
+
+	// Create organization and workspace
+	orgPath, _ := framework.NewOrganizationFixture(t, server)
+	partitionClusterPath, _ := framework.NewWorkspaceFixture(t, server, orgPath, framework.WithName("partitionset"))
+
+	cfg := server.BaseConfig(t)
+	kcpClusterClient, err := kcpclientset.NewForConfig(cfg)
+	require.NoError(t, err, "failed to construct kcp cluster client for server")
+	partitionSetClient := kcpClusterClient.TopologyV1alpha1().PartitionSets()
+
+	t.Logf("Creating a partitionSet not matching any shard")
+	partitionSet := &topologyv1alpha1.PartitionSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "test-partitionset",
+		},
+		Spec: topologyv1alpha1.PartitionSetSpec{
+			Dimensions: []string{"region"},
+			ShardSelector: &metav1.LabelSelector{
+				MatchExpressions: []metav1.LabelSelectorRequirement{
+					{
+						Key:      "excluded",
+						Operator: metav1.LabelSelectorOpDoesNotExist,
+					},
+				},
+			},
+		},
+	}
+	partitionSet, err = partitionSetClient.Cluster(partitionClusterPath).Create(ctx, partitionSet, metav1.CreateOptions{})
+	require.NoError(t, err, "error creating partitionSet")
+	framework.Eventually(t, func() (bool, string) {
+		partitionSet, err = partitionSetClient.Cluster(partitionClusterPath).Get(ctx, partitionSet.Name, metav1.GetOptions{})
+		require.NoError(t, err, "error retrieving partitionSet")
+		if conditions.IsTrue(partitionSet, topologyv1alpha1.PartitionSetValid) && conditions.IsTrue(partitionSet, topologyv1alpha1.PartitionsReady) {
```

Comment:
Roll this up in the reconciler to a ready condition please, waiting on more than one is fragile

Reply:
this mechanism is tested here.